ecosystem/typescript/sdk/SPEC.md

```diff
@@ -0,0 +1,210 @@
+- [ ] **2.2** The SDK **should** interact directly with the Aptos Indexer API.
+  - [ ] **2.2.1** The SDK **must** allow to pass `offset` and `limit` parameters to Indexer API request.
+  - [ ] **2.2.2** The SDK **must** validate the account address when interacting with the Indexer API.
+    - [ ] **2.2.2.1** The SDK **must** validate an account address is a hex string format.
+    - [ ] **2.2.2.2** The SDK **must** validate an account address is 66 chars long.
+    - [ ] **2.2.2.3** The SDK **must** padding an account address with leading `0`.
+  - [ ] **2.2.3** The SDK **should** handle premade GraphQL queries https://cloud.hasura.io/public/graphiql?endpoint=https://indexer.mainnet.aptoslabs.com/v1/graphql.
+  - [ ] **2.2.3** The SDK **should** allow the user to pass in a custom GraphQL query.
+- [ ] **2.3** The SDK **should** support interactions with Aptos devnet, testnet and mainnet networks.
+- [ ] **2.4** The SDK **should** support interactions with a custom URL.
+
+### 3. Account Management
+
+- [ ] **3.1** The SDK **must** support account creation.
+- [ ] **3.2** The SDK **must** support account balance read.
+- [ ] **3.3** The SDK **must** support key management for Ed25519 keys.
+- [ ] **3.4** The SDK **must** support local single-threaded sequence number management.
+- [ ] **3.5** The SDK **should** provide multi-agent signer support.
+- [ ] **3.6** The SDK **should** provide multi-Ed25519 signer support.
+- [ ] **3.7** The SDK **should** provide mnemonic support.
+- [ ] **3.8** The SDK **should** provide multi-sig support.
+- [ ] **3.9** The SDK **should** provide resource account support.
+
+### 4. Transaction
+
+- [ ] **4.1** The SDK **must** support entry function payload transaction submission.
+- [ ] **4.2** The SDK **must** support script payload transaction submission.
+- [ ] **4.3** The SDK **must** use the gas estimation API to determine the gas price when building transaction payloads.
+  - [ ] **4.3.1** The SDK **should** cache the response from the gas estimation API for a set period of time (e.g. 1 minute).
+- [ ] **4.4** The SDK **must** support simulation of transactions.
+
+### 5. Coin Management
+
+- [ ] **5.1** The SDK **must** support APT coin transfer.
+- [ ] **5.1** The SDK **should** support other coin transfer.
+
+## Maintenance Requirements
+
+### 6. Source Control
+
+- [ ] **6.1** The source code for the SDK **must** be maintained within Git version control.
+- [ ] **6.2** The source code **must** be hosted publicly.
+- [ ] **6.3** Development of new features **should** happen on feature branches.
+- [ ] **6.4** Feature branches **should** pass all tests and linting before they can be merged into the `main` branch.
+- [ ] **6.5** Source control **should** contain tags for each release of the SDK.
+- [ ] **6.6** The `main` branch **should** be kept in a condition that allows for direct use through checkout.
+- [ ] **6.7** The source code **should** use GitHub for public hosting.
```

Comment:
Also, it should have CICD testing to ensure it still works properly

Reply:
https://github.com/aptos-labs/aptos-core/pull/8107/files#diff-66f82625c7f6d4c059797eef86f6d7be1de7d4fe3edc783009b480a0f4eb5137R131